webapp/src/main/resources/template.soy

```diff
@@ -0,0 +1,16 @@
+{namespace validator.templates}
+
+/**
+ * Renders a basic explanation of the validation notice.
+ * @param numNotices
+ * @param notices
+ */
+{template .unknownColumnNotice}
+  <p class="warning">Warning - Unknown Column(s) found!</p>
+  <p><b>{$numNotices}</b> unknown column(s) found in:</p>
+  {foreach $notice in $notices}
+    <p>Filename: <i>{$notice.filename}</i></p>
+    <p>Column name: <i>{$notice.fieldName}</i> at position <i>{$notice.index}</i></p>
+  {/foreach}
+  <p>Please delete or rename column!</p>
+{/template}
```

Comment:
Do we need numNotices to be a separate param here, or could we get the number of notices from notices.length or something?

Reply:
It can be either. Would it be faster to use notices.Length or just read it in as a parameter? The report.json file already has numNotices as a variable.

Reply:
Is the numNotices in the json file the total number of notices? Or the number of unknownColumnNotices? I would guess it's probably the former, in which case it probably isn't what we want to put into the function.

Either way, I would say using notices.Length is probably pretty quick (likely constant time or at worst linear time, and we probably have only a few notices), in which case I would personally lean towards 'cleaner' code over performance optimisation.

Reply:
The numNotices in the json file is for the total number of unknownColumnNotices (or the number of notices for a certain notice code) using notices.size().

Is cleaner code using notices.Length or just using the parameters?

Reply:
I would use `notices.length` in the Soy template so you only need to pass `notices` into the generated JS function.

Reply:
Yep ok, I'll change that. Thanks!

Reply:
Done!